src/Stoa.ts

```diff
@@ -23,14 +24,85 @@ class Stoa {
+
+                this.ledger_storage.getValidatorsAPI(height, null, (err: Error | null, rows: any[]) =>
+                {
+                    if (err != null)
+                    {
+                        console.error("Failed to data lookup to the DB: " + err);
+                        res.status(500).send("Failed to data lookup");
+                        return;
+                    }
+
+                    if (rows.length)
+                    {
+                        let out_put:Array<ValidatorData> = new Array<ValidatorData>();
+
+                        for (const row of rows)
+                        {
+                            let preimage: IPreimage = {distance: row.distance, hash: '0'} as IPreimage;
+                            var validator: ValidatorData | null =
+                                new ValidatorData(row.address, row.enrolled_at, row.stake, preimage);
+                            out_put.push(validator);
+                        }
+                        res.status(200).send(JSON.stringify(out_put));
+                    }
+                    else
+                    {
+                        res.status(200).send("no data");
+                    }
+                });
         });
 
+        /**
+         * When a request is received through the `/validators/:address` handler
+         * Validators based on the block height if there is a height.
+         * If height null the most up to date state is expected.
+         * If an address exists, respond to the verifier data of that address.
+         */
         this.stoa.get("/validator/:address",
             (req: express.Request, res: express.Response, next: express.NextFunction) => {
             res.send(req.params.address + ':' + req.query.height);
+
+            var height: number = Number(req.query.height);
+            var address: string = String(req.params.address);
+
+            this.ledger_storage.getValidatorsAPI(height, address, (err: Error | null, rows: any[]) =>
+            {
+                if (err != null)
+                {
+                    console.error("Failed to data lookup to the DB: " + err);
+                    res.status(500).send("Failed to data lookup");
+                    return;
+                }
+
+                if (rows.length)
+                {
+                    let out_put:Array<ValidatorData> = new Array<ValidatorData>();
+
+                    for (const row of rows)
+                    {
+                        let preimage: IPreimage = {distance: row.distance, hash: '0'} as IPreimage;
+                        var validator: ValidatorData | null =
+                            new ValidatorData(row.address, row.enrolled_at, row.stake, preimage);
+                        out_put.push(validator);
+                    }
+                    res.status(200).send(JSON.stringify(out_put));
+                }
+                else
+                {
+                    res.status(200).send("no data");
```

Comment:
Does it really make sense that we return 200 and a random string as the result? When designing APIs think about what the client code will look like. You will end up having to check for this string.

Reply:
Right. I change the code now `res.status(204).send();`